src/Model/Casts/DateTime/DateTimeCast.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Mongolid\Model\Casts\DateTime;
+
+use DateTime;
+use MongoDB\BSON\UTCDateTime;
+use Mongolid\Util\LocalDateTime;
+
+class DateTimeCast extends BaseDateTimeCast
+{
+    /**
+     * @param UTCDateTime|null $value
+     * @return DateTime|null
+     */
+    public static function get(mixed $value): mixed
```

Comment:
If the return it's a `DateTime|null`, why doesn't set on the return type?
```suggestion
    public static function get(mixed $value): DateTime|null
```

Reply:
Because we are implementing an Interface on this class. 

We cannot explicitly return `?DateTime` because that would break the Interfaces contract and the Interface cannot return `?DateTime` because it must have a generic return in order to be flexible: our intentions are to implement multiple casts options, like Enums or even CustomCasts just as  Eloquent does.

Reply:
é possível especializar os retornos em uma interface.

Reply:
Tentei aqui e deu FatalError: 

![image](https://github.com/leroy-merlin-br/mongolid/assets/9309745/6a4c97fb-800e-4904-a2c2-e810a08878b1)

Reply:
Mesmo especializando só o retorno, também dá FatalError: 
![image](https://github.com/leroy-merlin-br/mongolid/assets/9309745/02461c7c-f412-430d-8f64-b482e7abf41f)

Reply:
Altera só o tipo do retorno, exemplo funfando:

https://onlinephp.io/c/9a66f2e8-3525-424c-b08d-3db286c9f5db